cpp/src/groupby/sort/group_argmin.cu

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <groupby/sort/group_single_pass_reduction_util.cuh>
+
+#include <thrust/transform.h>
+
+namespace cudf {
+namespace experimental {
+namespace groupby {
+namespace detail {
+
+std::unique_ptr<column> group_argmin(
+    column_view const& values,
+    size_type num_groups,
+    rmm::device_vector<size_type> const& group_labels,
+    column_view const& key_sort_order,
+    rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream)
+{
+  auto indices = type_dispatcher(
+    values.type(), reduce_functor<aggregation::ARGMIN>{},
+    values, num_groups, group_labels, mr, stream);
+
+  // The functor returns the index of minimum in the sorted values.
+  // We need the index of minimum in the original unsorted values.
+  // So we transform it with the sort order used to sort `values`.
+  // The values are transformed in-place so the validity is not changed.
+  mutable_column_view indices_view(indices->mutable_view());
+  thrust::transform(rmm::exec_policy(stream)->on(stream),
+    indices_view.begin<size_type>(), indices_view.end<size_type>(),
+    indices_view.begin<size_type>(),
+    [sort_order = key_sort_order.data<size_type>()] __device__ (size_type arg) {
+      return sort_order[arg];
```

Comment:
This needs to be changed.

Reply:
```suggestion
      return arg!=ARGMIN_SENTINEL ? sort_order[arg] : ARGMIN_SENTINEL;
```

Reply:
On second thought, this is just a gather. Why not use `gather`? It already has out of bounds checking.

Reply:
Neat.